src/tools/daxio/daxio.c

```diff
@@ -384,11 +396,19 @@ setup_device(struct ndctl_ctx *ndctl_ctx, struct daxio_device *dev, int is_dst)
 	if (!dev->is_devdax)
 		return 0;
 
-	if (is_dst) {
+	if (is_dst && !do_not_clear_bb) {
 		/* XXX - clear only badblocks in range bound by offset/len */
 		if (os_dimm_devdax_clear_badblocks_all(dev->path)) {
-			ERR("failed to clear badblocks on \"%s\"\n",
-					dev->path);
+			ERR("failed to clear bad blocks on \"%s\"\n"
+			    "       Probably you have not enough permissions to do that.\n"
```

Comment:
shouldn't we check errno first? :)

Reply:
ndctl sets wrong errno here, this is fixed in ndctl v63